cmake/Macros.cmake

```diff
@@ -281,11 +300,21 @@ macro(sfml_add_example target)
         add_executable(${target} ${target_input})
     endif()
 
+    # enable C++17 support
+    target_compile_features(${target} PUBLIC cxx_std_17)
+
     if (SFML_USE_STATIC_STD_LIBS)
         set_property(TARGET ${target} PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
     endif()
 
+    # enable precompiled headers
+    if (SFML_ENABLE_PCH)
+        message(VERBOSE "enabling PCH for SFML example '${target}'")
+        target_precompile_headers(${target} REUSE_FROM sfml-system)
+    endif()
+
     set_target_warnings(${target})
+    set_public_symbols_hidden(${target})
```

Comment:
Why do we need to hide symbols in executable targets? I thought that was only meaningful for library.

Reply:
PCH reuse requires the flags to be exactly the same for all projects using the same PCH.